boskos/handlers/server_client_test.go

```diff
@@ -33,11 +36,42 @@ import (
 	"k8s.io/test-infra/boskos/ranch"
 )
 
-func makeTestBoskos(r *ranch.Ranch) *httptest.Server {
-	handler := NewBoskosHandler(r)
+func makeTestBoskos(t *testing.T, r *ranch.Ranch) *httptest.Server {
+	handler := &testMuxWrapper{t: t, ServeMux: NewBoskosHandler(r)}
 	return httptest.NewServer(handler)
 }
 
+type testMuxWrapper struct {
```

Comment:
All of this feels pretty clunky, but the only thing I could find in the stdlib is the httptest.Recorder and that needs to be plugged into each handler. I'd appreciate any pointers if this can be simplified.